### DISCUSSION TITLE
AutoCompleteBox expose TextBox MaxLength

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#14560] [Ideas]
I need to set the maximum length of the `AutoCompleteBox` text, to do this I do it like this

```xaml
    <AutoCompleteBox SelectedItem="{...}">
      <AutoCompleteBox.Styles>
        <Style Selector="AutoCompleteBox /template/ TextBox#PART_TextBox">
          <Setter Property="MaxLength" Value="25"/>
        </Style>
      </AutoCompleteBox.Styles>
    </AutoCompleteBox>
```

I have multiple `AutoCompleteBoxes`, each with different `MaxLength`. So I'm forced to repeat the style for each of them.

**Comment by timunie** [comment]
I believe you can also write a Behavior or attached property (my favorite)

**Reply by workgroupengineering** [reply]
> I believe you can also write a Behavior or attached property (my favorite)

I did not think about it. But wouldn't turning TextBox.MaxLength into an attached property be a breaking-change?

**Reply by timunie** [reply]
Not in Avalonia, on your side. my:AutoCompletHelper.MaxLength
